kong-ngx-build

```diff
@@ -421,32 +426,32 @@ show_usage() {
 
 notice() {
   builtin echo -en "\033[1m"
-  echo "NOTICE: $@"
+  >&2 echo "NOTICE: $@"
   builtin echo -en "\033[0m"
 }
 
 succ() {
   builtin echo -en "\033[1;32m"
-  echo "SUCCESS: $@"
+  >&2 echo "SUCCESS: $@"
   builtin echo -en "\033[0m"
 }
 
 warn() {
   builtin echo -en "\033[1;33m"
-  echo "WARN: $@"
+  >&2 echo "WARN: $@"
   builtin echo -en "\033[0m"
 }
 
 fatal() {
   builtin echo -en "\033[1;31m"
-  echo "FATAL: $@"
+  >&2 echo "FATAL: $@"
   builtin echo -en "\033[0m"
   exit 1
 }
 
 err() {
   builtin echo -en "\033[1;31m"
-  echo "ERR: $@"
+  >&2 echo "ERR: $@"
```

Comment:
Why not also redirecting the color codes to stderr? It doesn't feel like this change accounts for situations in which stdout and stderr are split.

I suggest using `-t` (`man 1 test`) to check if stderr is a tty, and only output those color codes if that is the case. Output redirected to files doesn't need color.

Reply:
I think for the time being I will leave the non verbose output for another PR. Totally missed the color echo, good point.